src/app/(authenticated)/(dashboard)/[publicId]/settings/profile/page.tsx

```diff
@@ -4,81 +4,237 @@ import Image from "next/image";
 import { useForm } from "react-hook-form";
 import { Input } from "@/components/ui/input";
 import { Button } from "@/components/ui/button";
-import { useSession } from "next-auth/react";
+import { z } from "zod";
+import { Form, FormItem, FormLabel, FormMessage } from "@/components/ui/form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { api } from "@/trpc/react";
+import React, { useEffect, useRef, useState } from "react";
+import { useToast } from "@/components/ui/use-toast";
 
-import {
-  Form,
-  FormControl,
-  FormField,
-  FormItem,
-  FormLabel,
-  FormMessage,
-} from "@/components/ui/form";
+const profileSettingsSchema = z.object({
```

Comment:
Move this schema to a dedicated file so that it can be used on the backend as well.